lib/shared/permissions.js

```diff
@@ -90,68 +97,61 @@ exports.isElevatedUnixSync = () => {
   return (process.geteuid() === UNIX_SUPERUSER_USER_ID)
 }
 
-/**
- * @summary Get environment command prefix
- * @function
- * @private
- *
- * @param {Object} environment - environment map
- * @returns {String[]} command arguments
- *
- * @example
- * const commandPrefix = permissions.getEnvironmentCommandPrefix({
- *   FOO: 'bar',
- *   BAR: 'baz'
- * });
- *
- * childProcess.execSync(_.join(_.concat(commandPrefix, [ 'mycommand' ]), ' '));
- */
-exports.getEnvironmentCommandPrefix = (environment) => {
-  const isWindows = os.platform() === 'win32'
-
-  if (_.isEmpty(environment)) {
-    return []
-  }
+const escapeSh = (value) => {
+  // Make sure it's a string
+  // Replace ' -> '\'' (closing quote, escaped quote, opening quote)
+  // Surround with quotes
+  return `'${String(value).replace(/'/g, "'\\''")}'`
+}
 
-  const argv = _.flatMap(environment, (value, key) => {
-    if (_.isNil(value)) {
-      return []
-    }
+const escapeParamCmd = (value) => {
+  // Make sure it's a string
+  // Escape " -> \"
+  // Surround with double quotes
+  return `"${String(value).replace(/"/g, '\\"')}"`
+}
 
-    if (isWindows) {
-      // Doing something like `set foo=bar &&` (notice
-      // the space before the first ampersand) would
-      // cause the environment variable's value to
-      // contain a trailing space.
-      // See https://superuser.com/a/57726
-      return [ 'set', `${key}=${value}&&` ]
-    }
+const setEnvVarSh = (value, name) => {
+  return `export ${name}=${escapeSh(value)}`
+}
 
-    return [ `${key}=${value}` ]
-  })
+const setEnvVarCmd = (value, name) => {
+  return `set "${name}=${String(value)}"`
+}
 
-  if (isWindows) {
-    // This is a trick to make the binary afterwards catch
-    // the environment variables set just previously.
-    return _.concat(argv, [ 'call' ])
+// Exported for tests
+exports.createLaunchScript = (command, argv, environment) => {
+  const isWin = os.platform() === 'win32'
```

Comment:
Might be nice to change `isWin` to `isWindows`, to match code elsewhere?

Reply:
updated